Std/Data/BitVec/Lemmas.lean

```diff
@@ -468,3 +468,45 @@ protected theorem lt_of_le_ne (x y : BitVec n) (h1 : x <= y) (h2 : ¬ x = y) : x
   let ⟨y, lt⟩ := y
   simp
   exact Nat.lt_of_le_of_ne
+
+/-- For 'x <= 2^w', 'x / 2^(w - 1)' is 0 if 'x < 2^w', 1 otherwise -/
+theorem Nat.div_two_pow_pred {w : Nat} (hx : x < 2 ^ w) :
+    x / 2 ^ (w - 1) = if 2 ^ (w - 1) <= x then 1 else 0 := by
+  by_cases (2 ^ (w - 1) ≤ x)
+  case pos h =>
+    simp [h]
+    rw [Nat.div_eq_sub_div, Nat.div_eq_of_lt]
+    · apply Nat.sub_lt_right_of_lt_add h
+      rw [← Nat.mul_two, ← Nat.pow_succ]
+      rcases w with rfl | w' <;> simp_all
+      · apply Nat.lt_trans hx Nat.one_lt_two
+    · apply Nat.lt_of_lt_of_le
+      apply Nat.zero_lt_one
+      apply Nat.two_pow_pos
+    · exact h
+  case neg h =>
+    simp only [h, ↓reduceIte]
+    exact Nat.div_eq_of_lt (Nat.not_le.mp h)
```

Comment:
Does this not belong in `Data.Nat`?

Reply:
I don't feel so, it's a local theorem which is specialized to `BitVec`